### PR TITLE
[Blog] added blog about Unit/Integration tests for mesheryctl

### PR DESCRIPTION
**Pull request by Mr-DG-Wick**
Signed-off-by: Mr-DG-Wick <deepakgdkg1g8868@gmail.com>

**Description**

Blog : _How to write Unit/Integration Tests for mesheryctl_ by Piyush Singariya

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit fed3d5590973f70231034da3d14f040b0be64e42 at: https://61c346565caa393ec245c180--layer5ng.netlify.app

**Comment by leecalcote** [comment]
This is pretty great. Good work, guys.

File: src/collections/blog/2021-12-22-how-to-write-Unit-Integeration-Tests-for-mesheryctl/index.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            <li>Writing Integration Tests is however a tough task to tackle, but yeah that’s the fun part, running and making sure a command that’s hard to test, and making it automatic through GH workflows.</li>
+        </ol>
+
+<p>One most important note-</p>
+
+<p>Writing Tests and making them work is in itself a tough task to do, so even writing one test for a single function counts and makes a big difference over the period, so do not hesitate, start writing tests. Writing One test is enough to raise a PR and to get yourself started with Testing.</p>
+
+<p>The impact that you make with these tests is very high, I can’t even explain in words (I don’t have that good of a vocabulary). Your Tests are going to run in each PR raised for Meshery, for each commit someone is pushing to the PR, so if someone is damaging the supposed behavior of code, your tests is going to catch them red-handed, how cool is that?</p>
+
+<p>And yeah if you’re reading this, make sure to raise a PR with Tests for mesheryctl and add me to review those! Let’s get mesheryctl that 100% code-coverage badge.</p>
```
Comment (debo19) [inline src/collections/blog/2021-12-22-how-to-write-Unit-Integeration-Tests-for-mesheryctl/index.md:75]:
```suggestion
<p>If you’re reading this, make sure to raise a PR with Tests for mesheryctl and add me to review those! Let’s get mesheryctl that 100% code-coverage badge.</p>

```

File: src/collections/blog/2021-12-22-how-to-write-Unit-Integeration-Tests-for-mesheryctl/index.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            <li>So we run the command to start Meshery in GH workflow and after running the subcommand through the Tests we make sure that Meshery actually started. And if we successfully checked it, boom we aced writing Integration Tests.</li>
+            <li>Writing Integration Tests is however a tough task to tackle, but yeah that’s the fun part, running and making sure a command that’s hard to test, and making it automatic through GH workflows.</li>
+        </ol>
+
+<p>One most important note-</p>
+
+<p>Writing Tests and making them work is in itself a tough task to do, so even writing one test for a single function counts and makes a big difference over the period, so do not hesitate, start writing tests. Writing One test is enough to raise a PR and to get yourself started with Testing.</p>
+
+<p>The impact that you make with these tests is very high, I can’t even explain in words (I don’t have that good of a vocabulary). Your Tests are going to run in each PR raised for Meshery, for each commit someone is pushing to the PR, so if someone is damaging the supposed behavior of code, your tests is going to catch them red-handed, how cool is that?</p>
```
Comment (debo19) [inline src/collections/blog/2021-12-22-how-to-write-Unit-Integeration-Tests-for-mesheryctl/index.md:73]:
```suggestion
<p>The impact that you make with these tests is very high, I can’t even explain in words (I don’t have that good of a vocabulary). Your Tests are going to run in each PR raised for Meshery, for each commit someone is pushing to the PR, so if someone is damaging the supposed behavior of code, your tests are going to catch them red-handed, how cool is that?</p>

```

File: src/collections/blog/2021-12-22-how-to-write-Unit-Integeration-Tests-for-mesheryctl/index.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+---
+title: "How to write Unit/Integration Tests for mesheryctl"
+date: 2021-12-22 19:32:05 -0530
+author: Piyush Singariya
+thumbnail: ./Integration-Test.png
+category: "Meshery"
+tags:
+ - Projects
+ - Meshery
```
Comment (debo19) [inline src/collections/blog/2021-12-22-how-to-write-Unit-Integeration-Tests-for-mesheryctl/index.md:9]:
```suggestion
 - Meshery
 - mesheryctl

```

**Review by debo19 — COMMENTED** [review]
Looks good 🎉 , suggested some minor changes

**Comment by Mr-DG-Wick** [comment]
Thanks for the review @debo19!

**Comment by l5io** [comment]
🚀 Preview for commit cfb26b7398a776a3e4fe8f82a822eddff0499663 at: https://61c4677942fc117946b5aa98--layer5ng.netlify.app

**Comment by piyushsingariya** [comment]
@Mr-DG-Wick can you take care of failing build?

**Comment by l5io** [comment]
🚀 Preview for commit ab7f4cbb3f59082e0b259d4629921ecb5c4f7046 at: https://61c5eea8375345b3de0bd209--layer5ng.netlify.app

**Comment by Mr-DG-Wick** [comment]
Thank you @warunicorn19 😄

**Comment by warunicorn19** [comment]
No problem 😅

**Pull request merged**